substrate/pkg/controller/substrate/cluster/config.go

```diff
@@ -250,6 +273,47 @@ func defaultClusterConfig(substrate *v1alpha1.Substrate) *kubeadm.InitConfigurat
 	return defaultStaticConfig
 }
 
+func (c *Config) ensureAuthenticatorConfig(ctx context.Context, substrate *v1alpha1.Substrate) error {
+	identity, err := c.STS.GetCallerIdentityWithContext(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return fmt.Errorf("getting caller identity, %w", err)
+	}
+	configMap, err := iamauthenticator.Config(ctx, substrate.Name, substrate.Namespace,
+		aws.StringValue(discovery.Name(substrate, tenantControlPlaneNodeRole)), aws.StringValue(identity.Account))
+	if err != nil {
+		return err
+	}
+	logging.FromContext(ctx).Infof("Created config map for authenticator")
+	configDir := path.Join(ClusterCertsBasePath, aws.StringValue(discovery.Name(substrate)), authenticatorConfigDir)
+	if err := os.MkdirAll(configDir, 0700); err != nil {
+		return fmt.Errorf("failed to create directory, %w", err)
+	}
+	if err := ioutil.WriteFile(path.Join(configDir, "config.yaml"), []byte(configMap.Data["config.yaml"]), 0644); err != nil {
+		return fmt.Errorf("writing authenticator config, %w", err)
+	}
+	return nil
+}
+
+func (c *Config) staticPodSpecForAuthenticator(ctx context.Context, substrate *v1alpha1.Substrate) error {
+	podTemplateSpec := iamauthenticator.PodSpec(func(spec v1.PodSpec) v1.PodSpec {
```

Comment:
Shouldn't this be a pod template spec, so that you can modify everything you need in one go? Seems like you're getting in your own way on line 304.